### PR TITLE
fix truss push --promote

### PR DESCRIPTION
**Pull request by rcano-baseten**
## :rocket: What
* `truss push --promote` doesn't work when the model hasn't been pushed before 
* This PR ensures that the flow is unblocked

### CODE REVIEW

File: truss/remote/baseten/core.py [hunk 1]
```diff
@@ -271,7 +272,7 @@ def create_truss_service(
         return model_version_json["id"], model_version_json["version_id"]
 
     if model_id is None:
-        if environment:
+        if environment != PRODUCTION_ENVIRONMENT_NAME:
```
Comment (jrochette) [inline truss/remote/baseten/core.py:275]: nit: can you add unit tests for this method?

**Review by jrochette — APPROVED** [review]

**Review by spal1 — APPROVED** [review]

File: truss/remote/baseten/core.py (hunk 1 above)
Comment (spal1) [inline truss/remote/baseten/core.py:275]: is it possible for `environment` to be `None` here? we should handle that as well, maybe something like 
```suggestion
        if environment and environment != PRODUCTION_ENVIRONMENT_NAME:
```

Comment (rcano-baseten) [reply truss/remote/baseten/core.py]: done!

Comment (rcano-baseten) [reply truss/remote/baseten/core.py]: yeah good catch. I saw that as I was writing tests as well. I think we should be good now.

**Comment by spal1** [comment]
🚢 tested the flows that broke previously:
```
truss push empty --model-name HUH --promote
? 🎮 Which remote do you want to connect to? baseten-staging-1-25
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
✨ Model HUH was successfully pushed ✨
Your Truss has been deployed into the production environment. After it successfully deploys, it will become the next production deployment of your
model.
```
```
truss chains push hello_chain/hello.py --promote
? 🎮 Which remote do you want to connect to? baseten-staging-1-25
Using chain workspace dir: `/Users/sam/hello_chain` (files under this dir will be included as dependencies in the remote deployments and are
importable).
Generating truss chainlet model for `AnotherDependency`.
Truss not found in pip requirements, auto-adding: `truss==0.9.43`.
`promote=True` overrides `publish` to `True`.
Pushing chainlet `AnotherDependency-7860bada` as a truss model on Baseten (publish=False, promote=True).
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Pushed `AnotherDependency`
Generating truss chainlet model for `RandInt`.
Truss not found in pip requirements, auto-adding: `truss==0.9.43`.
`promote=True` overrides `publish` to `True`.
Pushing chainlet `RandInt-c936b01b` as a truss model on Baseten (publish=False, promote=True).
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Pushed `RandInt`
Generating truss chainlet model for `HelloWorld_10_10`.
Truss not found in pip requirements, auto-adding: `truss==0.9.43`.
`promote=True` overrides `publish` to `True`.
Pushing chainlet `HelloWorld_10_10-0af50979` as a truss model on Baseten (publish=False, promote=True).
Compressing... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Uploading... ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 100% 0:00:00
Pushed `HelloWorld_10_10`
```

**Pull request merged**